src/interfaces.rs

```diff
@@ -233,23 +233,19 @@ impl Interfaces {
                     .get_mapping(interface_name, interface_path)
                     .ok_or_else(|| {
                         AstarteError::ReceiveError(format!(
-                            "Mapping '{}' doesn't exist",
-                            interface_path
+                            "Mapping '{interface_path}' doesn't exist",
                         ))
                     })?;
 
-                match mapping {
-                    crate::interface::Mapping::Datastream(_) => {}
-                    crate::interface::Mapping::Properties(map) => {
-                        if bdata.is_empty() && !map.allow_unset {
-                            return Err(AstarteError::ReceiveError(
-                                "Do not unset a mapping without allow_unset".into(),
-                            ));
-                        }
+                if let crate::interface::Mapping::Properties(map) = mapping {
+                    if (individual == AstarteType::Unset) && !map.allow_unset {
+                        return Err(AstarteError::ReceiveError(
+                            "Do not unset a mapping without allow_unset".into(),
+                        ));
                     }
                 }
 
-                if individual != mapping.mapping_type() {
+                if (individual != AstarteType::Unset) && (individual != mapping.mapping_type()) {
                     return Err(AstarteError::ReceiveError(
                         "You are receiving the wrong type for this mapping".into(),
                     ));
```

Comment:
What do you think if you rewrite something  like this?
``` Rust
if (individual == AstarteType::Unset) {
    if bdata.is_empty() && !map.allow_unset {
        return Err(AstarteError::ReceiveError( "Do not unset a mapping without allow_unset".into(),
        ));
    }
}
```
the validation of a property with `unset` fails only if  ` bdata.is_empty() && !map.allow_unset` , while is true in the other case, also the line 252 can remain unchanged.

Reply:
By reverting the guard: 
```
if (individual != AstarteType::Unset) && (individual != mapping.mapping_type()) {
```
to:
```
if individual != mapping.mapping_type() {
```
Unsetting a property would always fail as the mapping does not match.
Furthermore, the condition `bdata.is_empty()` is actually equivalent to  `individual == AstarteType::Unset`. As `individual` is set to `AstarteType::Unset` in the function `deserialize`.